File: src/utils/logger.ts
type Level = 'debug' | 'info' | 'warn' | 'error'

function write (level: Level, message: string): void {
  const timestamp = new Date().toISOString()
  process.stderr.write(`[${timestamp}] ${level.toUpperCase()} ${message}\n`)
}

export const logger = {
  debug: (message: string): void => { write('debug', message) },
  info: (message: string): void => { write('info', message) },
  warn: (message: string): void => { write('warn', message) },
  error: (message: string): void => { write('error', message) }
}

File: internal/logger/logger.go
package logger

import (
	"fmt"
	"os"
	"time"
)

func write(level, message string) {
	timestamp := time.Now().UTC().Format("2006-01-02T15:04:05.000Z")
	fmt.Fprintf(os.Stderr, "[%s] %s %s\n", timestamp, level, message)
}

func Debug(message string) { write("DEBUG", message) }
func Info(message string)  { write("INFO", message) }
func Warn(message string)  { write("WARN", message) }
func Error(message string) { write("ERROR", message) }

func Infof(format string, args ...any)  { Info(fmt.Sprintf(format, args...)) }
func Warnf(format string, args ...any)  { Warn(fmt.Sprintf(format, args...)) }
func Errorf(format string, args ...any) { Error(fmt.Sprintf(format, args...)) }
